### PR TITLE
Fix rename for containers on swarm network

### PR DESCRIPTION
**Pull request by aboch**
- The rename logic has not been updated yet to work in swarm overlay networks.
  This PR takes care of it.

Signed-off-by: Alessandro Boch <aboch@docker.com>

### CODE REVIEW

File: endpoint.go [hunk 1]
```diff
@@ -576,39 +577,68 @@ func doUpdateHostsFile(n *network, sb *sandbox) bool {
...
-	if !ok {
-		return fmt.Errorf("watch null for network %q", n.Name())
+	if c.isAgent() {
+		if err = ep.deleteServiceInfoFromCluster(); err != nil {
+			return types.InternalErrorf("Could not delete service state for endpoint %s from cluster on rename: %v", ep.Name(), err)
+		}
+		// Add enough delay to avoid the above delete event be suppressed
+		// in favor of the subsequent add event while queued to be gossiped
+		// to the other nodes in the cluster
+		time.Sleep(50 * time.Millisecond)
```
Comment (aboch) [inline endpoint.go]: The sleep can be removed if #1689  is accepted and merged

Comment (mavenugo) [inline endpoint.go]: Agreed. I think we should remove this in favor of #1689

Comment (sanimej) [inline endpoint.go]: I think this case is problematic. We can't assume the delete and create will be delivered in order, in that case remote nodes will end up adding the new IP, but not remove the old IP.

Comment (aboch) [inline endpoint.go]: good point, the msgs will be sent in order, but could be delivered in reverse order.
In that case, we may still need to keep this sleep.
At the end it is for a very specific rename case only.

**Pull request merged**